internal/cmdrender/executor.go

```diff
@@ -77,24 +84,39 @@ func (e *Executor) Execute(ctx context.Context) error {
 		return err
 	}
 
-	pkgWriter := &kio.LocalPackageReadWriter{PackagePath: string(root.pkg.UniquePath)}
-
 	// format resources before writing
 	_, err = filters.FormatFilter{UseSchema: true}.Filter(resources)
 	if err != nil {
 		return err
 	}
 
-	err = pkgWriter.Write(resources)
-	if err != nil {
-		return fmt.Errorf("failed to save resources: %w", err)
+	if e.Output == nil {
```

Comment:
We can add a comment marking this block as `in-place` along with some comments suggested below.

Reply:
Added.